src/lib/encoder.ts

```diff
@@ -352,21 +366,90 @@ const valueTypeEncodingMap = {
     },
   },
   uint256: {
-    encode: (value: string | number) =>
-      abiCoder.encodeParameter('uint256', value),
-    decode: (value: string) => abiCoder.decodeParameter('uint256', value),
+    encode: (value: string | number) => {
+      return abiCoder.encodeParameter('uint256', value);
+    },
+    decode: (value: string) => {
+      if (!isHex(value)) {
+        throw new Error(`Can't convert ${value} to uint256, value is not hex.`);
+      }
+
+      const numberOfBytes = countNumberOfBytes(value);
+
+      if (numberOfBytes > 32) {
+        throw new Error(
+          `Can't convert hex value ${value} to uint256. Too many bytes. ${numberOfBytes} is above the maximal number of bytes 32`,
+        );
+      }
+
+      return BigNumber(value).toNumber();
+    },
   },
   bytes32: {
-    encode: (value) => abiCoder.encodeParameter('bytes32', value),
+    encode: (value: string | number) => {
+      let valueToEncode: string;
+
+      if (typeof value === 'string' && !isHex(value)) {
+        // if we receive a plain string (e.g: "hey!"), convert it to utf8-hex data
+        valueToEncode = toHex(value);
+      } else if (typeof value === 'number') {
+        // if we receive a number as input, convert it to hex
+        valueToEncode = numberToHex(value);
+      } else {
+        valueToEncode = value;
+      }
+
+      const numberOfBytes = countNumberOfBytes(valueToEncode);
+
+      // CHECK that there is not more than 4 bytes
+      if (numberOfBytes > 32) {
```

Comment:
4 bytes or 32 bytes?

Reply:
I think the comment is useless anyway:


this
```
if (numberOfBytes > 32)
```

does not require a comment to tell what it does.